### ISSUE TITLE
bug: Safari doesn't render wheel .picker-before and after gradients in IonDatetime when used inside of an IonModal

### ISSUE THREAD

**Issue by puopg** [ionic-team/ionic-framework#25770]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

When using the @ionic/react package, the `<IonDatetime presentation="time">` component renders as expected, it appears in a wheel picker form. Chrome no issues so far.

On Desktop Safari however, when used inside of an `IonModal`, the styles don't seem to be honored, and the background css that should be applied for the picker before and picker after gradients are not rendered by Safari.

https://user-images.githubusercontent.com/7741481/185013577-628932c8-a595-47b6-8d39-87f04b1d7882.mov

### Codesandbox of this issue:
https://codesandbox.io/s/ionic-6-long-press-item-list-forked-ytbxgt?file=/src/App.tsx

Could it be a webkit and shadow dom thing?

### Expected Behavior

Expecting same rendering behavior when used anywhere.

### Steps to Reproduce

For the codesandbox, it should render the date picker on the screen, and then when the modal opens, a datepicker inside the modal. You can see the differences.

### Code Reproduction URL

https://codesandbox.io/s/ionic-6-long-press-item-list-forked-ytbxgt?file=/src/App.tsx

### Ionic Info

@ionic/react: 6.2.2
Safari: Version 15.5 (17613.2.7.1.8)

**Comment by puopg** [comment]
Doing a little more research this evening.

So I tried hard coding some inline styles into the before and after element. At first I applied the same styles that ionic is applying, but nothing happened. But applying rules that don't involve css variables, seems to work, since I just replaced all the css variables with like `red` haha.

**Comment by puopg** [comment]
Update:

So inline still has some issues I think, however, I've been seeing differences between:

```
linear-gradient(to bottom, #FFFFFF 20%, rgba(255, 255, 255, 0.8) 100%)
```

and
 
```
linear-gradient(to bottom, #FFFFFF 20%, red 100%)
```

Without any rgba, it works! So I wonder if theres some weirdness when the browser is rendering opacity?

**Comment by puopg** [comment]
Promoting the picker-before and after to their own stacking context also seems to work

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am going to close this as a duplicate of https://github.com/ionic-team/ionic-framework/issues/25755. A fix for this was released today as part of Ionic v6.2.3.

**Comment by puopg** [comment]
@liamdebeasi 

Confirmed, 6.2.3 fixes this: 

New codesandbox:
https://codesandbox.io/s/ionic-6-long-press-item-list-forked-piuhi7?file=/package.json

I will note though, that seems if you declare in package.json

@ionic/react: "6.2.2"
@ionic/react-router: "^6.0.0" (which prolly grabs 6.2.3)

the issue also doesnt appear, which makes it seem like im using 6.2.2, but really the fix is coming in with the @ionic/react-router package.
